LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,347 @@
+    private var initialTherapySettings: TherapySettings
+    var therapySettings: TherapySettings
+
+    public init(therapySettings: TherapySettings) {
+        self.therapySettings = therapySettings
+        self.initialTherapySettings = therapySettings
+    }
+    
+    /// Reset to original
+    func reset() {
+        therapySettings = initialTherapySettings
+    }
+}
+
+public struct TherapySettingsView: View, HorizontalSizeClassOverride {
+    @Environment(\.dismiss) var dismiss
+
+    weak var delegate: TherapySettingsViewDelegate?
+    
+    @ObservedObject var viewModel: TherapySettingsViewModel
+
+    @State var isEditing: Bool = false
+    
+    // Hack to test edit button below section or next to header
+    public enum DesignVersion {
+        case A, B, C
+        func cycle() -> DesignVersion {
+            switch self {
+            case .A: return .B
+            case .B: return .C
+            case .C: return .A
+            }
+        }
+    }
+    @State var editButtonDesignVersion: DesignVersion = .C
+
+    private let mode: PresentationMode
+    
+    public init(mode: PresentationMode = .modal, editButtonDesignVersion: DesignVersion = .C, viewModel: TherapySettingsViewModel) {
+        self.mode = mode
+        self.viewModel = viewModel
+        self.editButtonDesignVersion = editButtonDesignVersion
+    }
+    
+    public var body: some View {
+        switch mode {
+        case .modal: return AnyView(navigationContent)
+        case .flow: return AnyView(content)
+        }
+    }
+    
+    private var navigationContent: some View {
+        return NavigationView {
+            content
+        }
+    }
+    
+    private var content: some View {
+        List {
+            correctionRangeSection
+            temporaryCorrectionRangesSection
+        }
+        .listStyle(GroupedListStyle())
+        .navigationBarTitle(Text(NSLocalizedString("Therapy Settings", comment: "Therapy Settings screen title")))
+        .navigationBarItems(leading: backOrCancelButton, trailing: editOrDoneButton)
+        .navigationBarBackButtonHidden(isEditing)
+        .environment(\.horizontalSizeClass, horizontalOverride)
+    }
+}
+
+typealias HKQuantityGuardrail = Guardrail<HKQuantity>
+
+extension TherapySettingsView {
+    
+    private var therapySettings: TherapySettings {
+        viewModel.therapySettings
+    }
+    
+    private var glucoseUnit: HKUnit? {
+        therapySettings.glucoseTargetRangeSchedule?.unit
+    }
+    
+    private var backOrCancelButton: some View {
+        return Button<AnyView>( action: {
+            if self.isEditing {
+                // TODO: confirm
+                self.delegate?.cancel()
+                self.viewModel.reset()
+                self.isEditing.toggle()
+            } else {
+                self.dismiss()
+            }
+        }) {
+            if isEditing {
+                return AnyView(Text(NSLocalizedString("Cancel", comment: "Cancel button text")))
```

Comment:
[nit] I'm pretty sure you can just do `LocalizedString` in SwiftUI

Reply:
`LocalizedString` is not from SwiftUI... it is in `LoopKit`, but you are right.  `LocalizedString` is a way to grab localized assets from the Framework's bundle, and since this UI is in `LoopKitUI`, localizations should probably come from it. 
However, looking through the code, I notice this is not consistent, so I imagine some cleanup will be in order.